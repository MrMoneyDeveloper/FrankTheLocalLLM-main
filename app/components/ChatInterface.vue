<template>
  <div class="flex flex-col h-full">
    <div ref="container" class="flex-1 overflow-y-auto p-4 space-y-2">
      <div v-for="(m, i) in messages" :key="i" :class="m.role === 'user' ? 'text-right' : 'text-left'">
        <span :class="m.role === 'user' ? 'bg-blue-200' : 'bg-gray-200'" class="inline-block px-2 py-1 rounded">{{ m.content }}</span>
      </div>
    </div>
    <div class="p-2 border-t">
      <input v-model="message" @keyup.enter="send" placeholder="Ask something..." class="border w-full p-2" />
    </div>
  </div>
</template>
<script setup>
import { ref, nextTick } from 'vue'

const message = ref('')
const messages = ref([])
const container = ref(null)
const API_BASE = (import.meta.env?.VITE_API_BASE || '/api').replace(/\/$/, '')


async function send () {
  if (!message.value) return
  const text = message.value
  messages.value.push({ role: 'user', content: text })
  message.value = ''
  await nextTick(() => { container.value.scrollTop = container.value.scrollHeight })
  const resp = await fetch(`${API_BASE}/chat`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ message: text })
  })
  const data = await resp.json()
  messages.value.push({ role: 'bot', content: data.response })
  await nextTick(() => { container.value.scrollTop = container.value.scrollHeight })
}
</script>
