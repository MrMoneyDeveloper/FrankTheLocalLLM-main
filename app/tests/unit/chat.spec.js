import { mount } from '@vue/test-utils'
import ChatInterface from '../../components/ChatInterface.vue'
import { describe, it, expect, vi } from 'vitest'
import flushPromises from 'flush-promises'

global.fetch = vi.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ response: 'hi there' })
  })
)

describe('ChatInterface', () => {
  it('sends message and shows reply', async () => {
    const wrapper = mount(ChatInterface)
    await wrapper.find('input').setValue('hello')
    await wrapper.vm.send()
    await flushPromises()
    expect(fetch).toHaveBeenCalled()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
