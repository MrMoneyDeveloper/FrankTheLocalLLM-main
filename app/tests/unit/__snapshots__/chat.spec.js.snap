// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatInterface > sends message and shows reply 1`] = `
"<div class="flex flex-col h-full">
  <div class="flex-1 overflow-y-auto p-4 space-y-2">
    <div class="text-right"><span class="bg-blue-200 inline-block px-2 py-1 rounded">hello</span></div>
    <div class="text-left"><span class="bg-gray-200 inline-block px-2 py-1 rounded">hi there</span></div>
  </div>
  <div class="p-2 border-t"><input placeholder="Ask something..." class="border w-full p-2"></div>
</div>"
`;
